### PR TITLE
fix(ui5-view-settings-dialog): improve settings management

### PR DESCRIPTION
**Pull request by tsanislavgatev**
fixes: https://github.com/SAP/ui5-webcomponents/issues/7752

### CODE REVIEW

File: packages/fiori/src/ViewSettingsDialog.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
 import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
 import { isPhone } from "@ui5/webcomponents-base/dist/Device.js";
 import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
-import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import UI5Element, { ChangeInfo } from "@ui5/webcomponents-base/dist/UI5Element.js";
```
Comment (hinzzx) [inline packages/fiori/src/ViewSettingsDialog.ts:9]: maybe **import type { ChangeInfo }...**, on the next row, would be better for consistency in my opinion ?

**Review by nnaydenow — CHANGES_REQUESTED** [review]
https://sap.github.io/ui5-webcomponents/nightly/playground/?path=/docs/docs-development-understanding-components-metadata--docs#slot-configuration-settings

Could you please check invalidateOnChildChange option for the slot decorators?

File: packages/fiori/src/ViewSettingsDialog.ts [hunk 2]
```diff
@@ -241,6 +242,8 @@ class ViewSettingsDialog extends UI5Element {
 	@property({ type: Boolean, noAttribute: true })
 	_filterStepTwo!: boolean;
 
+	_slotChanged!: boolean;
```
[the remark this answers is not in the document]
Comment (tsanislavgatev) [reply packages/fiori/src/ViewSettingsDialog.ts:245]: yes, missed when changing approach.

**Comment by tsanislavgatev** [comment]
> https://sap.github.io/ui5-webcomponents/nightly/playground/?path=/docs/docs-development-understanding-components-metadata--docs#slot-configuration-settings
> 
> Could you please check invalidateOnChildChange option for the slot decorators instead of using the onInvalidation hook?

I've tried it but in this case it doesn't work as we have validation between confirmed/current state. The different states are because after close we keep the selection and when opened again we restore the selected state. Only with invalidation, it would still return the old state, that's why we overwrite the variable and we replace with the latest state.

**Review by nnaydenow — APPROVED** [review]

**Review by hinzzx — APPROVED** [review]

**Comment by lateefsofi** [comment]
I don't see this fixing https://github.com/SAP/ui5-webcomponents/issues/7752 issue, can you please check it again

**Pull request merged**